### PR TITLE
Add `amd-aie-direct` HAL target (3/n)

### PR DESCRIPTION
**Pull request by makslevental**
This PR is part of a stack of PRs that refactor the dependency on MLIR-AIE. See https://github.com/nod-ai/iree-amd-aie/issues/430 for more information.

This PR adds a new target `amd-aiet-direct`, directly references AIEVec passes (for eventual lowering of core code to chess LLVM IR) and also (simultaneously) vendors `XCLBinGen.cpp` from `mlir-aie` (second part used to be broken out in https://github.com/nod-ai/iree-amd-aie/pull/422 but on @newling's suggestion I squashed). Note, `XCLBinGen` and `AIETargetDirect` are basically carbon copies of what currently exists except `AIETargetDirect` doesn't shell out and calls `aie2xclbin` directly. Note also that `XCLBinGen` will undergo further revision up to for the chess backend (but that'll have to wait until I get home).

To test, I added some lines to `print_ir_aie2xclbin.sh`. I'm happy to bike shed this but probably not too hard.

### CODE REVIEW

**Comment by makslevental** [comment]
@newling @nirvedhmeshram this is now ready for your keen eyes!

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.h
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.h:43]: nit: newline

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:1]: I think if the file was first introduced in 2023, we should leave this as 2023?

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.h
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.h:1]: same here.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp:339]: can we just use `entryPointNamesFb[ordinal]` instead of FOO?

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp:340]: can we stringify the ordinal into a hex string here? (It will help when we start merging kernels)
here is what I was thinking, Something like this has worked for me.

```

    // Convert ordinal to hexadecimal string for xclbin kernel id.
    std::stringstream ss;
    ss << "0x" << std::hex << ordinal;
    std::string ordinalHex = ss.str();
```

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp:283]: This is a bug I introduced sorry,

https://github.com/nod-ai/iree-amd-aie/pull/453

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/PluginRegistration.cpp
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/PluginRegistration.cpp:1]: 2023 here too?

**Review by nirvedhmeshram — COMMENTED** [review]
Thanks, looks good, just some minor comments.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp
Comment (makslevental) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:1]: TIL

**Review by nirvedhmeshram — APPROVED** [review]
LGTM, after fixing the test/CI.

**Review by newling — APPROVED** [review]
Some minor comments but they can all be ignored as I think it'll easier to make changes once everything is landed, and temporary code is removed.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp
Comment (makslevental) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp:47]: This flag is in `AIETarget` and so this target should also support it but I can't register the `cl::opt` (because they're static globals). So I left it here as a reminder to myself.

**Pull request merged**
